### PR TITLE
kernel: have k_sched_lock call _sched_lock

### PR DESCRIPTION
**Pull request by tyrantlizard77**
Having two implementations of the same thing is bad,
especially when one can just call the other inline version.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Review by lpereira — APPROVED** [review]
I'm curious why this was done this way in the first place.

**Review by andyross — APPROVED** [review]
makes sense.  I'm betting it evolved this way because there was some assembly at some point that poked it.

The scheduler lock is a really subtle thing, not a lock at all really but an optimization.  It prevents code that would otherwise context switch from doing so "right now" (i.e. "not yet!  things aren't set up right"), and relies on the code using it to call _Swap() manually at the right time.  We don't use it much, and it's not IMHO a good candidate for a driver API at all.

Basically I wouldn't cry if an equivalent fix was to k_sched_lock() from the API entirely and expose it as an internal to the one extra-kernel spot (ARM IRQ offload) that uses it.

**Review by nashif — APPROVED** [review]
